### PR TITLE
Realm role empty state

### LINKED ISSUE #176: Add empty state to realm roles section
Extends #135 ... when the only remaining role is deleted from the roles list, this action should trigger an empty state.

### PR DESCRIPTION
**Pull request by jenny-s51**
## Motivation

Closes #176  

## Checklist:

- [x] Code has been tested locally by PR requester
- [x] User-visible strings are using the react-i18next framework (useTranslation)
- [x] Help has been implemented
- [ ] axe report has been run and resulting a11y issues have been resolved
- [x] Unit tests have been created/updated
- [x] Formatting has been performed via prettier/eslint
- [x] Type checking has been performed via 'yarn check-types'

### CODE REVIEW

File: src/realm-roles/RealmRolesSection.tsx [hunk 1]
```diff
@@ -63,6 +59,14 @@ export const RealmRolesSection = () => {
           >
             <RolesList roles={roles} refresh={loader} />
           </PaginatingTableToolbar>
+        ) : (
```
Comment (mfrances17) [inline src/realm-roles/RealmRolesSection.tsx:62]: This looks fine, but I'm wondering what the purpose is of NoRealmRoles.tsx, which is an empty state for realm roles. It seems like we should choose one or the other, unless I am misunderstanding the purpose of each? They seem redundant.

**Review by jenny-s51 — COMMENTED** [review]
Nice catch @mfrances17! I started on that file before @srambach added `ListEmptyState`. We don't need `NoRealmRoles.tsx` anymore so I've deleted it.

**Review by ssilvert — APPROVED** [review]
Looks good.  Merging.

**Review by mfrances17 — COMMENTED** [review]
LGTM

**Pull request merged**